### PR TITLE
Filter challenges

### PR DESCRIPTION
**Pull request by Paajt**
https://github.com/user-attachments/assets/676941e4-1716-4859-a776-2a8ff00db9e6

Load API with a button for testing visually in browser. (Will be removed once connected to other parts of website).
All styling or lack of styling in video is just for test purposes in browser environment.

Reseted challenges.html to same version as in main to avoid merge conflicts (challenges.html used for filter testing saved localy for now).

Filters added (works in synchronization to combine multiple filters):
- [x] Online/On-site sorting with checkboxes.
- [x] Ratings: From min 1-5 to max 1-5 (radio buttons).
- [x] Tags/labels: Clicking a tag keeps it selected with that filter until clicked again to remove tag and filter.
- [x] Search: User input to filter through titles and descriptions.

Todo:
- [ ] Reset filters button to remove all filters (Group descision).
- [ ] Investigate if trim should be used on user input.
- [ ] Connect all of this to rest of the project.
- [ ] Remove a few console.log() that seems unnecessary in the end product.

### CODE REVIEW

**Review by alfredbrannare — APPROVED** [review]
Well done, looks great!

File: challenges.html [hunk 1]
```diff
@@ -206,6 +206,7 @@ <h5>Follow us on</h5>
     <script src='filter.js'></script>
     <script src="load-challenges.js"></script>
     <script src="modal.js"></script>
+    <script src="challengesFilter.js"></script>
```
Comment (Paajt) [inline challenges.html:209]: Merge conflict resolved 👍

Comment (itsJoel59) [reply challenges.html:209]: Great job man 🙂

**Review by itsJoel59 — APPROVED** [review]
Green light from me 🚦

**Comment by NordicNomadicLife** [comment]
Looks good its good to go for me 👍

**Pull request merged**